pkg/go.mod

```diff
@@ -5,32 +5,124 @@ go 1.24.3
 replace github.com/akuity/kargo/api => ../api
 
 require (
+	github.com/akuity/kargo v1.7.2
```

Comment:
The way things were before: `akuity/kargo` depends on `akuity/kargo/pkg` and `akuity/kargo/api`. `akuity/kargo/pkg` depends only on `akuity/kargo/api`.

I think this line hints that something snuck into `akuity/kargo/pkg` that is importing something from `akuity/kargo`. i.e. Some lower level thing is now importing some higher level thing. (This seems circular and I'm surprised `go mod` didn't totally barf on this.)

Why it matters: I'm sure that this was accidental, but it probably dragged along a lot of indirect dependencies that `akuity/kargo/pkg` does not otherwise need.

I haven't pinpointed the offending import yet.

Reply:
I’ll look for it. I honestly thought it would have barfed too, which is why I didn’t notice this

Reply:
Found it: the `expressions/function` dir is pulling from `internal` (which I also thought wouldn't work). Namely

```
"github.com/akuity/kargo/internal/controller/freight"
"github.com/akuity/kargo/internal/kargo"
```